IHP/Controller/FileUpload.hs

```diff
@@ -49,12 +49,16 @@ import qualified System.Process as Process
 --
 -- See 'filesByName' if multiple files can be uploaded by the user in a single form submission.
 --
--- See 'IHP.FileStorage.ControllerFunctions.storeFile' to upload the file to S3 or similiar cloud storages.
+-- See 'IHP.FileStorage.ControllerFunctions.storeFile' to upload the file to S3 or similar cloud storages.
 --
 fileOrNothing :: (?context :: ControllerContext) => ByteString -> Maybe (FileInfo LBS.ByteString)
 fileOrNothing !name =
         case ?context.requestContext.requestBody of
-            FormBody { files } -> lookup name files
+            FormBody { files } ->
+                -- Search for the file, and confirm it's not an empty one.
+                case lookup name files of
+                    Just file | fileContent file /= "" -> Just file
```

Comment:
Performance wise it's better to use a `not (null fileContent)` here instead, like in ` Just fileInfo | not (LBS.null (fileInfo.fileContent))`

```suggestion
                    Just file | fileContent file /= "" -> Just file
```